woocommerce/woocommerce
# Marketplace: Implement the subscriptions table row

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
- Add table loading animation using `TablePlaceholder`
- Implement components for cells in a row
    - product name
    - status
    - expiry
    - Auto-renew information
    - Version number
    - Activate/Install
    - Actions

### How to test the changes in this Pull Request:

1. Checkout this branch and build WooCommerce Admin assets (`cd plugins/woocommerce-admin && pnpm run start`)
2. Make sure you connect a WooCommerce.com account that has some subscriptions
3. Visit **WooCommerce > Extensions > My Subscriptions**
4. See the three tables: Installed on this store, Not in use, and Free to install. ~~The Free to install table will be missing data, and it will be fixed later. Please see the logged issue on the project board.~~ The free to install table is now gone.
    - [ ] Remove free to install table (@raicem)
6. See the table row that has new components. This now has outdated designs because we made some tweaks. Please see the logged issue on the project board.
    - So there is not much to validate in terms of designs. But we can review the code as it'll lay the foundation for later changes.
7. Click on the **Install** button on the **Not in use** table and confirm it installs the plugin (basically re-testing #40630 so that nothing is broken)
8. Please test around this issue.

**Screenshot**
![image](https://github.com/woocommerce/woocommerce/assets/10389957/ec19e58f-6a8f-44d4-9d88-786afd310ca1)
